clientv3/watch.go

```diff
@@ -643,12 +646,7 @@ func (w *watchGrpcStream) serveSubstream(ws *watcherStream, resumec chan struct{
 	// lazily send cancel message if events on missing id
 }
 
-func (w *watchGrpcStream) newWatchClient() (pb.Watch_WatchClient, error) {
-	// connect to grpc stream
-	wc, err := w.openWatchClient()
-	if err != nil {
-		return nil, v3rpc.Error(err)
-	}
+func (w *watchGrpcStream) newWatchClient() (ws pb.Watch_WatchClient, err error) {
```

Comment:
@heyitsanthony This functions is pretty fat right now. I feel we should rename this to newOrResumeWatchClient. Or we can separate the new, resume case.